### ISSUE TITLE
Infinite raft loop

### ISSUE THREAD

**Issue by slayer** [influxdb/influxdb#1241]
After server restart, influxdb does not start properly - it does not listen web:

Log:
```
[2014/12/17 00:35:02 MSK] [INFO] (github.com/influxdb/influxdb/datastore.(*ShardDatastore).GetOrCreateShard:158) DATASTORE: opening or creating shard /opt/influxdb/shared/data/db/shard_db_v2/
[2014/12/17 00:35:02 MSK] [INFO] (github.com/influxdb/influxdb/datastore.(*ShardDatastore).GetOrCreateShard:158) DATASTORE: opening or creating shard /opt/influxdb/shared/data/db/shard_db_v2/
[2014/12/17 00:35:02 MSK] [INFO] (github.com/influxdb/influxdb/coordinator.(*RaftServer).startRaft:409) Recovered from log
[2014/12/17 00:35:02 MSK] [INFO] (github.com/influxdb/influxdb/server.(*Server).ListenAndServe:96) Waiting for local server to be added
[2014/12/17 00:35:02 MSK] [INFO] (github.com/influxdb/influxdb/wal.(*WAL).SetServerId:109) Setting server id to 1 and recovering
[2014/12/17 00:35:02 MSK] [DEBG] (github.com/influxdb/influxdb/wal.(*WAL).recover:503) Finished wal recovery
[2014/12/17 00:35:03 MSK] [INFO] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftEventHandler:448) (raft:7b0fa960d53dfc97) Selected as leader. Starting leader loop.
[2014/12/17 00:35:04 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:05 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:06 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:07 MSK] [INFO] (github.com/influxdb/influxdb/server.(*Server).ListenAndServe:112) Sending change connection string command (server12352:8099,server14789:8099) (http://server
[2014/12/17 00:35:07 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:08 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:09 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:10 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:11 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:12 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:13 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:14 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:15 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:16 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:17 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:18 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:19 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
[2014/12/17 00:35:20 MSK] [DEBG] (github.com/influxdb/influxdb/coordinator.(*RaftServer).raftLeaderLoop:467) (raft:7b0fa960d53dfc97) Executing leader loop.
...
```

lsof:
```
influxdb 10190 influxdb  cwd    DIR   254,1     4096      2 /
influxdb 10190 influxdb  rtd    DIR   254,1     4096      2 /
influxdb 10190 influxdb  txt    REG   254,1 47158958 128368 /opt/influxdb/versions/0.8.7/influxdb
influxdb 10190 influxdb  mem    REG   254,1  1603600   7980 /lib/x86_64-linux-gnu/libc-2.13.so
influxdb 10190 influxdb  mem    REG   254,1    89056   1238 /lib/x86_64-linux-gnu/libgcc_s.so.1
influxdb 10190 influxdb  mem    REG   254,1   131107   7976 /lib/x86_64-linux-gnu/libpthread-2.13.so
influxdb 10190 influxdb  mem    REG   254,1    66824    300 /lib/x86_64-linux-gnu/libbz2.so.1.0.4
influxdb 10190 influxdb  mem    REG   254,1    92752   4600 /lib/x86_64-linux-gnu/libz.so.1.2.7
influxdb 10190 influxdb  mem    REG   254,1   530736   7984 /lib/x86_64-linux-gnu/libm-2.13.so
influxdb 10190 influxdb  mem    REG   254,1    31744   7995 /lib/x86_64-linux-gnu/librt-2.13.so
influxdb 10190 influxdb  mem    REG   254,1   136936   7977 /lib/x86_64-linux-gnu/ld-2.13.so
influxdb 10190 influxdb    0r   CHR     1,3      0t0   1205 /dev/null
influxdb 10190 influxdb    1w   CHR     1,3      0t0   1205 /dev/null
influxdb 10190 influxdb    2w   CHR     1,3      0t0   1205 /dev/null
influxdb 10190 influxdb    3w   REG   254,1    51238 238672 /opt/influxdb/shared/log.txt
influxdb 10190 influxdb    4w   REG   254,1    11551 134065 /opt/influxdb/shared/data/db/shard_db_v2/00005/LOG
influxdb 10190 influxdb    5u  IPv6 1001593      0t0    TCP *:8090 (LISTEN)
influxdb 10190 influxdb    6u  0000     0,9        0   1203 anon_inode
influxdb 10190 influxdb    7r   DIR   254,1     4096 134050 /opt/influxdb/shared/data/db/shard_db_v2/00005
influxdb 10190 influxdb    8uW  REG   254,1        0 134070 /opt/influxdb/shared/data/db/shard_db_v2/00005/LOCK
influxdb 10190 influxdb    9u   REG   254,1       92 139640 /opt/influxdb/shared/data/db/shard_db_v2/00005/000122.log
influxdb 10190 influxdb   10r   REG   254,1      502 134148 /opt/influxdb/shared/data/db/shard_db_v2/00005/000120.sst
influxdb 10190 influxdb   11u   REG   254,1      448 134150 /opt/influxdb/shared/data/db/shard_db_v2/00005/MANIFEST-000121
influxdb 10190 influxdb   12w   REG   254,1    11527 139642 /opt/influxdb/shared/data/db/shard_db_v2/00003/LOG
influxdb 10190 influxdb   13r   DIR   254,1     4096 134075 /opt/influxdb/shared/data/db/shard_db_v2/00003
influxdb 10190 influxdb   14uW  REG   254,1        0 134102 /opt/influxdb/shared/data/db/shard_db_v2/00003/LOCK
influxdb 10190 influxdb   15u   REG   254,1       92 348576 /opt/influxdb/shared/data/db/shard_db_v2/00003/000107.log
influxdb 10190 influxdb   16r   REG   254,1      501 139682 /opt/influxdb/shared/data/db/shard_db_v2/00003/000105.sst
influxdb 10190 influxdb   17u   REG   254,1      283 139688 /opt/influxdb/shared/data/db/shard_db_v2/00003/MANIFEST-000106
influxdb 10190 influxdb   18w   REG   254,1    16157 348399 /opt/influxdb/shared/data/db/shard_db_v2/00004/LOG
influxdb 10190 influxdb   19r   DIR   254,1     4096 134026 /opt/influxdb/shared/data/db/shard_db_v2/00004
influxdb 10190 influxdb   20uW  REG   254,1        0 134048 /opt/influxdb/shared/data/db/shard_db_v2/00004/LOCK
influxdb 10190 influxdb   21u   REG   254,1       92 348580 /opt/influxdb/shared/data/db/shard_db_v2/00004/000118.log
influxdb 10190 influxdb   23u   REG   254,1      704 349099 /opt/influxdb/shared/data/db/shard_db_v2/00004/MANIFEST-000117
influxdb 10190 influxdb   28w   REG   254,1    11540 348402 /opt/influxdb/shared/data/db/shard_db_v2/00001/LOG
influxdb 10190 influxdb   29r   DIR   254,1     4096 133672 /opt/influxdb/shared/data/db/shard_db_v2/00001
influxdb 10190 influxdb   30uW  REG   254,1        0 134024 /opt/influxdb/shared/data/db/shard_db_v2/00001/LOCK
influxdb 10190 influxdb   31u   REG   254,1       92 348583 /opt/influxdb/shared/data/db/shard_db_v2/00001/000113.log
influxdb 10190 influxdb   32r   REG   254,1      502 348581 /opt/influxdb/shared/data/db/shard_db_v2/00001/000111.sst
influxdb 10190 influxdb   33r   REG   254,1  2123523 349103 /opt/influxdb/shared/data/db/shard_db_v2/00004/000119.sst
influxdb 10190 influxdb   34u   REG   254,1      367 349104 /opt/influxdb/shared/data/db/shard_db_v2/00001/MANIFEST-000112
influxdb 10190 influxdb   35u   REG   254,1    12921 134105 /opt/influxdb/shared/data/raft/log
influxdb 10190 influxdb   37r   REG   254,1  2059131 139684 /opt/influxdb/shared/data/db/shard_db_v2/00004/000120.sst
```

**Comment by clkao** [comment]
I am seeing similar issue - after influxdb 0.8.7 crashed with oom, restarting influxdb is showing "Executing leader loop" in logs and only port 8090 is being listened, no listeners on 8083/8086

**Comment by paramaggarwal** [comment]
I'm also experiencing this issue. InfluxDB 0.8.8.

**Comment by toddboom** [comment]
I've seen this occur in a few cases and it seems to be something related to LevelDB/RocksDB. It's generally proportional to the size of the database on disk, and has always resolved itself given enough time (i.e. InfluxDB will eventually start up if you wait long enough). It also seems to not be repeatable once the process has finished. Unfortunately, for now the only fix seems to be waiting.

I don't know if it's related to some sort of start-up compaction, or some other initialization process, but it's likely going to be gone in v0.9.0, since the datastore has been completely rewritten. I'm going to leave this open until we get some early builds of v0.9.0 available, but I'll close it out once initial testing is complete.

**Comment by slayer** [comment]
In my case it did not resolved in few days.
Thanks, waiting for 0.9.0

**Comment by m1keil** [comment]
@slayer in your log, I see two hosts:  (server12352:8099,server14789:8099)
Who's server12352 and who's server14789? Is it possible that servers's hostname was changed during reboot?

**Comment by slayer** [comment]
Yes, it is possible. Do you think problem in hostname inconsistency?

**Comment by m1keil** [comment]
Yes, see previous discussion [here](https://github.com/influxdb/influxdb/issues/1164#issuecomment-69759596).

**Comment by otoolep** [comment]
There is a completely new Raft implementation in 0.9.0. I'm not sure anything is going to happen here.

@toddboom ?

**Comment by pauldix** [comment]
yeah, this is a won't fix for previous versions and isn't a problem in 0.9.0
